src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/TextSpanIntervalIntrospector.cs

```diff
@@ -16,10 +16,17 @@ public int GetLength(TextSpan value)
         => value.Length;
 }
 
-internal sealed class TextSpanIntervalTree(IEnumerable<TextSpan> values)
+internal sealed class TextSpanIntervalTree(IEnumerable<TextSpan>? values)
     : SimpleIntervalTree<TextSpan, TextSpanIntervalIntrospector>(new TextSpanIntervalIntrospector(), values)
 {
-    public TextSpanIntervalTree(params TextSpan[] values) : this((IEnumerable<TextSpan>)values)
+    public TextSpanIntervalTree() : this(null)
```

Comment:
avoids two wasted allocs in the case where no values are initially provided to the tree.